File: src/main/java/com/sinnerschrader/s2b/accounttool/config/authentication/LdapAuthenticationDetailsSource.java
package com.sinnerschrader.s2b.accounttool.config.authentication;

import com.sinnerschrader.s2b.accounttool.config.ldap.LdapConfiguration;
import org.springframework.security.web.authentication.WebAuthenticationDetails;
import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;

import javax.servlet.http.HttpServletRequest;

public class LdapAuthenticationDetailsSource extends WebAuthenticationDetailsSource {

	private LdapConfiguration ldapConfiguration;

	public LdapAuthenticationDetailsSource(LdapConfiguration ldapConfiguration) {
		this.ldapConfiguration = ldapConfiguration;
	}

	@Override
	public WebAuthenticationDetails buildDetails(HttpServletRequest context) {
		return new LdapAuthenticationDetails(ldapConfiguration, context);
	}
}

File: src/main/java/com/sinnerschrader/s2b/accounttool/config/authentication/LdapAuthenticationDetailsSource.kt
package com.sinnerschrader.s2b.accounttool.config.authentication

import com.sinnerschrader.s2b.accounttool.config.ldap.LdapConfiguration
import org.springframework.security.web.authentication.WebAuthenticationDetailsSource
import javax.servlet.http.HttpServletRequest


class LdapAuthenticationDetailsSource(private val ldapConfiguration: LdapConfiguration) : WebAuthenticationDetailsSource() {
    override fun buildDetails(context: HttpServletRequest) = LdapAuthenticationDetails(ldapConfiguration, context)
}
